src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1085,6 +1085,7 @@ bool PolicyHandler::ReceiveMessageFromSDK(const std::string& file,
         correlation_id, vehicle_data_args, application_manager_);
   } else {
     LOG4CXX_WARN(logger_, "Exchange wasn't successful");
+    application_manager_.ProcessFailedPTU();
```

Comment:
@mked-luxoft I believe you can use `OnPTUFinished(ret)` to call `ProcessFailedPTU`/`ProcessSuccessfulStatusUpdate` in AM